packages/fxa-auth-server/lib/routes/cloud-tasks.ts

```diff
@@ -3,33 +3,108 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import isA from 'joi';
+import { Container } from 'typedi';
 import { ConfigType } from '../../config';
 import DESCRIPTION from '../../docs/swagger/shared/descriptions';
-import { ReasonForDeletion } from '../account-delete';
-import { AuthRequest } from '../types';
+import { AccountDeleteManager, ReasonForDeletion } from '../account-delete';
+import DB from '../db/index';
+import { AuthLogger, AuthRequest } from '../types';
 import validators from './validators';
+import { ERRNO } from '../error';
+import pushBuilder from '../push';
 
 export type DeleteAccountTaskPayload = {
   uid: string;
   customerId?: string;
   reason: ReasonForDeletion;
 };
+type FxaDbForCloudTask = Pick<
+  Awaited<ReturnType<ReturnType<typeof DB>['connect']>>,
+  'account' | 'devices'
+>;
+type PushForCloudTask = Pick<
+  ReturnType<typeof pushBuilder>,
+  'notifyAccountDestroyed'
+>;
+type Log = AuthLogger & { activityEvent: (data: Record<string, any>) => void };
 
 export class CloudTaskHandler {
-  private config: ConfigType;
+  private log: Log;
+  private db: FxaDbForCloudTask;
+  private push: PushForCloudTask;
+  private accountDeleteManager: AccountDeleteManager;
 
-  constructor(config: ConfigType) {
-    this.config = config;
+  constructor({
+    log,
+    db,
+    push,
+  }: {
+    log: Log;
+    db: FxaDbForCloudTask;
+    push: PushForCloudTask;
+  }) {
+    this.log = log;
+    this.db = db;
+    this.push = push;
+
+    this.accountDeleteManager = Container.get(AccountDeleteManager);
   }
 
-  deleteAccount(taskPayload: DeleteAccountTaskPayload) {
+  async deleteAccount(taskPayload: DeleteAccountTaskPayload) {
+    try {
+      // get account from MySQL
+      const account = await this.db.account(taskPayload.uid);
+
+      // We fetch the devices to notify before deleteAccount()
+      // because obviously we can't retrieve the devices list after!
+      const devices = await this.db.devices(taskPayload.uid);
+      const push = this.push;
+      const log = this.log;
+
+      const notify = async () => {
+        try {
+          log.info('accountDeleted.byCloudTask', account);
+          await push.notifyAccountDestroyed(taskPayload.uid, devices);
+          await log.notifyAttachedServices('delete', {} as AuthRequest, {
+            uid: taskPayload.uid,
+          });
+          // because a cloud task request is very different from a user request
+          // we cannot emit metrics in the same fashion
+          if (!account.metricsOptOutAt) {
```

Comment:
I'm going to double check that opting out of metrics should prevent account deletion logs.  That feels like it might be too far.  But I'll ask.

Reply:
Update: yes, we should be logging that accounts were deleted, regardless of opt-out status.

Reply:
> Update: yes, we should be logging that accounts were deleted, regardless of opt-out status.

@clouserw do you want me to change that for user requested deletes also?